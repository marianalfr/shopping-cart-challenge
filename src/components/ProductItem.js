import React from 'react';
import styled from 'styled-components';

// Styled Components -->

const Row = styled.div`
    position: relative;
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    margin-bottom: 32px;
`;

const ColProduct = styled.div`
    width: 45%;
    display: flex;
    align-items: center;
    flex-flow: row nowrap;
`;

const Image = styled.img`
    margin-right: 16px;
    width: 72px;
    height: 72px;
    border: 1px solid ${props => props.theme.color.lighter};
    border-radius: 4px;
`;

const Title = styled.h1`
    ${props => props.theme.title.product}
`;

const Code = styled.p`
    border-radius: 4px;
    color: ${props => props.theme.color.light};
    letter-spacing: 0.13px;
    font-weight: 400;
    font-size: 12px;
    line-height: 16px;
`;

const Col = styled.div`
    width: 20%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

const Button = styled.button`
    padding: 0 8px;
    height: 40px;
    border: none;
    background: transparent;
    color: ${props => props.theme.color.main};
    font-weight: bold;
    cursor: pointer;
    font-size: 20px;
    line-height: 25px;
`;

const Quantity = styled.input`
    width: 40px;
    height: 40px;
    border: 2px solid #dbdbe0;
    border-radius: 4px;
    text-align: center;
    font-size: 14px;
    line-height: 17px;
`;

const Span = styled.span`
    color: ${props => props.theme.color.darkest};
    font-size: 16px;
    margin-right: 4px;
`;

const ColTotal = styled.div`
    width: 15%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

// ProductItem Component -->

const ProductItem = props => {
    return(
        <Row>
            <ColProduct>
                <Image src={props.product.image} alt={props.product.product}/>
                <div>
                    <Title>{props.product.product}</Title>
                    <Code>Product Code {props.product.code}</Code>
                </div>
            </ColProduct>
            <Col>
                <Button>-</Button>
                <Quantity type="text" value={props.product.quantity}></Quantity>
                <Button>+</Button>
            </Col>
            <Col>
                <Span>{props.product.price}</Span>
                <Span>€</Span>
            </Col>
            <ColTotal>
                <Span>{props.product.price * props.product.quantity}</Span>
                <Span>€</Span>
            </ColTotal>
        </Row>
    )
} 

export default ProductItem;