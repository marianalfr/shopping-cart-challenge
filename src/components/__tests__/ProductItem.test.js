import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import ProductItem from '../ProductItem';

Enzyme.configure({ adapter: new Adapter() });

const product = {
    product: 'shirt',
    price: 20,
    quantity: 3,
    code: 'X7R2OPX',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
    images: {
        thumb: require('../../images/shirt.png'),
        large: require('../../images/shirt-large.jpg')
    },
    offer: {
        type: {
        category: 'percentage',
        name: '-5%',
        numbers: {
            percentage: 5
        }
        },
        minQty: 3 
    }
};

describe('ProductItem Component', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <ProductItem 
                product = { product }
                updateQuantity = { () => {} }
            />
        );
        expect(wrapper.debug()).toMatchSnapshot(); 
    });
});