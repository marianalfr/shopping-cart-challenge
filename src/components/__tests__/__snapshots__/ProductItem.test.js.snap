// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem Component renders correctly 1`] = `
"<Fragment>
  <styled.div>
    <styled.div>
      <styled.div />
      <styled.img src=\\"shirt.png\\" alt=\\"shirt\\" />
      <div>
        <styled.h1 onClick={[Function: toggleModal]}>
          shirt
        </styled.h1>
        <styled.p>
          Product Code 
          X7R2OPX
        </styled.p>
        <styled.p>
          -5%
           when buying 
          3
           or more.
        </styled.p>
      </div>
    </styled.div>
    <styled.div>
      <styled.button onClick={[Function: removeItem]}>
        -
      </styled.button>
      <styled.input type=\\"text\\" value={3} readOnly={true} />
      <styled.button onClick={[Function: addItem]}>
        +
      </styled.button>
    </styled.div>
    <styled.div>
      <styled.span>
        20
      </styled.span>
      <styled.span>
        €
      </styled.span>
    </styled.div>
    <styled.div>
      <styled.span>
        60
      </styled.span>
      <styled.span>
        €
      </styled.span>
    </styled.div>
  </styled.div>
</Fragment>"
`;
